src/Provider/ImageProvider.php

```diff
@@ -83,25 +83,40 @@ public function getHelperProperties(MediaInterface $media, $format, $options = [
         ];
 
         if (MediaProviderInterface::FORMAT_ADMIN !== $format) {
-            $srcSet = [];
 
-            foreach ($this->getFormats() as $providerFormat => $settings) {
-                // Check if format belongs to the current media's context
-                if (0 === strpos($providerFormat, $media->getContext())) {
-                    $width = $this->resizer->getBox($media, $settings)->getWidth();
+            $srcSetFormats = $this->getFormats();
 
-                    $srcSet[] = sprintf('%s %dw', $this->generatePublicUrl($media, $providerFormat), $width);
+            if (isset($options['srcset']) && is_array($options['srcset'])) {
```

Comment:
The optionsresolver could help with that kind of thing.

Reply:
Is this still something I should look into, given the answer on your other comment? I don't see an immediate benefit from implementing it now.